### PR TITLE
adds latency metrics

### PR DESCRIPTION
**Pull request by jmelis**
npm package:
https://www.npmjs.com/package/express-prom-bundle

### CODE REVIEW

**Comment by jmelis** [comment]
cc @aditya-konarde @rporres

**Comment by rporres** [comment]
jenkins build is failing: 
```
npm ERR! peer dep missing: prom-client@^12.0.0, required by express-prom-bundle@6.0.0
Traceback (most recent call last):
  File "/coreapi/main.py", line 168, in <module>
    main()
  File "/coreapi/main.py", line 152, in main
    request_id = run_sa_post(gateway_url, gateway_user_key)
  File "/coreapi/main.py", line 79, in run_sa_post
    if generate_deps_file(ecosystem, manifest) is None:
  File "/coreapi/main.py", line 35, in generate_deps_file
    out = sp.check_output(command, shell=True)
  File "/usr/lib64/python3.6/subprocess.py", line 356, in check_output
    **kwargs).stdout
  File "/usr/lib64/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command 'cd /coreapi/repo; npm install; npm list --prod --json > npmlist.json' returned non-zero exit status 1.
Build step 'Execute shell' marked build as failure
Setting status of 43cc0817cd73ccedf2c865ffdf1f646ffbba45dd to FAILURE with url https://ci.ext.devshift.net/job/app-sre-qontract-server-gh-pr-check/27/ and message: ' '
Using context: ci.ext.devshift.net PR build
Finished: FAILURE

File: src/server.ts [hunk 1]
```diff
@@ -8,12 +8,21 @@ import * as db from './db';
...
 
 import prometheusClient = require('prom-client');
+const promBundle = require("express-prom-bundle");
 
 interface IAcct {
   [key: string]: number;
 }
 
-// enable prom-client to expose default application metrics
+const metricsMiddleware = promBundle({
```
Comment (rporres) [inline src/server.ts:17]: I would use `formatStatusCode` to have status codes in the format of `2xx`, `5xx`, etc and avoid keeping status codes that we are not going to use.

Comment (jmelis) [reply src/server.ts:17]: totally, thanks for catching this!

File: src/server.ts [hunk 2]
```diff
@@ -8,12 +8,22 @@ import * as db from './db';
...
+  includeMethod: true,
+  includePath: true,
+  normalizePath: [
+    ['^/graphqlsha/.*', '/graphqlsha/#sha'],
+  ],
+  formatStatusCode: (res: express.Response) => Math.floor(res.statusCode / 100) + 'xx',
```
Comment (rporres) [inline src/server.ts:23]: nit: It looks it is going to do the right thing, but I can't help thinking it is very weird to do arithmetics with what should be a string containing the status code. Can't you just get the first char of the status code and add `xx`? 😂

Comment (rporres) [reply src/server.ts:23]: forget about it. The standard says they're actually integers, so it make sense to make arithmetics with them https://tools.ietf.org/html/rfc7231#page-47

**Comment by jmelis** [comment]
I have rebased a lot of things, so I'm closing this PR and submitting a new one.

**Pull request closed without merging**